baseplate/context/cassandra.py

```diff
@@ -148,8 +153,25 @@ def execute_async(self, query, parameters=None, timeout=_NOT_SET):
         future.add_errback(_on_execute_failed, span)
         return future
 
-    def prepare(self, query):
+    def prepare(self, query, cache=True):
+        """Prepares a CQL statement.
+
+        :param bool cache: If set to True (default), prepared statements will be
+        automatically cached and reused. The cache is keyed on the text of the
+        statement.
+        """
+
         trace_name = "{}.{}".format(self.context_name, "prepare")
+        query_hash = ""
+
         with self.server_span.make_child(trace_name) as span:
+            if cache:
+                query_hash = hashlib.md5(query.encode()).hexdigest()
+                if query_hash in self.prepared_cache:
+                    return self.prepared_cache[query_hash]
+
             span.set_tag("statement", query)
```

Comment:
Should I add a span for the hash lookup? Not sure how I should place that in here.

Reply:
That's probably too granular for an automatic metric.